client/types/data.d.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { StoreDescriptor } from '@wordpress/data/build-types/types';
+import { AnyConfig } from '@wordpress/data/src/types';
+
+declare module '@wordpress/data' {
+	type ControlDescriptor = {
+		type: string;
+		storeKey: string | StoreDescriptor< AnyConfig >;
+		actionName: string;
+		args: unknown[];
+	};
+
+	interface Controls {
+		dispatch: (
+			storeKey: string,
```

Comment:
I think it's the other way around

For the input, it's typed `string | StoreDescriptor< AnyConfig >` and for the output, it's a `string`